srml/session/src/lib.rs

```diff
@@ -359,10 +359,16 @@ impl<T: Trait> Module<T> {
 	}
 
 	/// Disable the validator of index `i`.
-	pub fn disable(i: usize) {
+	pub fn disable_index(i: usize) {
 		T::SessionHandler::on_disabled(i);
 		<Changed<T>>::put(true);
 	}
+
+	/// Disable the validator identified by `c`. (If using with the staking module, this would be
+	/// their *controller* account.)
+	pub fn disable(c: &T::AccountId) -> rstd::result::Result<(), ()> {
```

Comment:
Disabling is meant to be coupled on the consensus side with ignoring of validators' messages. 

This should be accompanied by some kind of signal in the header chain to make consensus engines aware that a validator has been disabled. One option I see is to have the `on_disabled` implementation do it. I'm fine with making that a follow-on, but we should have a clear idea of how it would fit in with these changes in `session`.

Reply:
yeah the idea is to have a separate PR that introduces support for `on_disable` to each of grandpa, babe and aura. that's out of my comfort zone though.

Reply:
i did put an entry for it in the spreadsheet:
![image](https://user-images.githubusercontent.com/138296/59549058-3fb17800-8f58-11e9-93a0-3c73d2589c05.png)